### PR TITLE
Improve perf when creating Item URLs

### PR DESCRIPTION
**Pull request by gsnedders**
Also fix JSON serialisation of reftests, which currently get serialised with their `url_base` (this bug goes back to the introduction of manifestv4).

### CODE REVIEW

File: tools/manifest/item.py [hunk 1]
```diff
@@ -84,7 +84,15 @@ def id(self):
 
     @property
     def url(self):
-        return urljoin(self.url_base, self._url)
+        # we can outperform urljoin, because we know we just have path relative URLs
+        if self._url.startswith("/"):
```
Comment (jgraham) [inline tools/manifest/item.py]: Is `self.url[0]` not going to be faster?

File: tools/manifest/item.py [hunk 2]
```diff
@@ -84,7 +84,15 @@ def id(self):
...
+            # TODO: MANIFEST6
+            # this is actually a bug in older generated manifests
+            return self._url
```
Comment (jgraham) [inline tools/manifest/item.py]: Also is this actually correct if the `url_base` isn't `/`?

Comment (gsnedders) [reply tools/manifest/item.py]: Yes, given resolving `/foo/bar` against `/_mozilla` is `/foo/bar`. Really the problem is the old manifests are broken…

File: tools/manifest/item.py (hunk 1 above)
Comment (gsnedders) [reply tools/manifest/item.py]: oh, probably, hopefully we never have empty strings!

File: tools/manifest/item.py [hunk 3]
```diff
@@ -84,7 +84,15 @@ def id(self):
...
+        elif self.url_base == "/":
+            return "/" + self._url
+        else:
+            return urljoin(self.url_base, self._url)
```
Comment (Ms2ger) [inline tools/manifest/item.py]: Nit: no else-after-return, please.

**Pull request merged**